ngraph_bridge/ngraph_utils.cc

```diff
@@ -463,6 +475,198 @@ bool IsProcessedByNgraphPass(Graph* g) {
+  // done creating event object
+
+  if (files.empty())  // create new tf event file
+  {
+    str_path += ("stats" + to_string(file_idx));
+
+    tensorflow::EventsWriter writer(str_path);
+
+    try {
+      writer.WriteEvent(event);
+    } catch (...) {
+      return errors::Internal("Error writing event to new event file");
+    }
+  } else  // append to tf event file in dir
+  {
+    std::string file_path = str_path + files[0];
+
+    // open tf event file at location file_path
+    std::unique_ptr<tensorflow::WritableFile> writable_file;
+
+    env->NewAppendableFile(file_path, &writable_file);
+    tensorflow::io::RecordWriter record_writer(writable_file.get());
+    // done opening tf event file at location file_path
+
+    try {
+      record_writer.WriteRecord(event.SerializeAsString());
+    } catch (...) {
+      return errors::Internal("Error updating event file with new event");
+    }
+  }
+
+  return Status::OK();
+}
+
+Status CreateSummaryFromGraph(tensorflow::Graph* graph,
+                              std::string filename_prefix) {
+  // convert Graph* to GraphDef*
+  tensorflow::GraphDef* gdef = nullptr;
+  graph->ToGraphDef(gdef);
+
+  // create event summary writer
+  tensorflow::EventsWriter writer(filename_prefix);
+  tensorflow::Event event;
+
+  // write graph to event summary
+  try {
+    event.set_graph_def(gdef->SerializeAsString());
+    writer.WriteEvent(event);
+  } catch (...) {
+    return errors::Internal("Error writing graph to event file");
+  }
+
+  return Status::OK();
+}
+
+Status CreateSummaryFromGraphDef(tensorflow::GraphDef* graph_def,
+                                 std::string filename_prefix) {
+  // create event summary writer
+  tensorflow::EventsWriter writer(filename_prefix);
+  tensorflow::Event event;
+
+  // write graph to event summary
+  try {
+    event.set_graph_def(graph_def->SerializeAsString());
+    writer.WriteEvent(event);
+  } catch (...) {
+    return errors::Internal("Error writing graph def to event file");
+  }
+
+  return Status::OK();
+}
+
+void AddSessionNameAttr(int file_idx, std::set<string> nodes, Graph* graph) {
+  for (auto node : graph->op_nodes()) {
+    if (node->type_string() == "NGraphEncapsulate") {
+      node->AddAttr(("_session_name" + to_string(file_idx)),
+                    GetSessionName(file_idx, nodes));
+    }
+  }
+}
+
+std::string GetSessionName(int file_idx, std::set<std::string> nodes) {
+  std::string name = "";
+
+  if (nodes.size() == 0) {
+    name = to_string(file_idx);
+  } else {
+    std::string entry;
+
+    for (auto it = nodes.begin(); it != nodes.end();
+         ++it)  // get last element in set
+    {
+      entry = *it;
+    }
+
+    int scope_idx = entry.find_first_of("/");
+
+    if (scope_idx != std::string::npos) {
+      name = to_string(file_idx) + "_" + entry.substr(0, scope_idx);
+    } else {
+      name = to_string(file_idx) + "_" + entry;
+    }
+  }
+
+  return name;
+}
+
+void MakeTBDir(std::string str_path) {
```

Comment:
try this: https://github.com/NervanaSystems/ngraph/blob/master/src/ngraph/file_util.hpp#L60

Then you don't have to write a new function